pdp/errors.yaml

```diff
@@ -988,3 +991,417 @@ errors:
+  msg: "Negative integer value %d underflows %s"
+  args:
+  - field: n
+  - field: v.Type()
+
+- id: requestUnmarshalFloatConstError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal float to unchengeable %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalFloatTypeError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal float to %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalAddressConstError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal address to unchengeable %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalAddressTypeError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal address to %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalNetworkConstError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal network to unchengeable %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalNetworkTypeError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal network to %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalDomainConstError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal domain to unchengeable %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalDomainTypeError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal domain to %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalSetOfStringsConstError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal set of strings to unchengeable %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalSetOfStringsTypeError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal set of strings to %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalSetOfNetworksConstError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal set of networks to unchengeable %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalSetOfNetworksTypeError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal set of networks to %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalSetOfDomainsConstError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal set of domains to unchengeable %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalSetOfDomainsTypeError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal set of domains to %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalListOfStringsConstError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal list of strings to unchengeable %s"
+  args:
+  - field: v.Type()
+
+- id: requestUnmarshalListOfStringsTypeError
+  fields:
+  - id: v
+    type: reflect.Value
+  msg: "Can't unmarshal list of strings to %s"
+  args:
+  - field: v.Type()
+
+- id: responseEffectError
+  fields:
+  - id: effect
+    type: int
+  msg: "Unknown effect %d"
+  args:
+  - field: effect
+
+- id: ResponseServerError
```

Comment:
The id `ReponseServerError` should be lowercase.
As it is, the id `ResponseServerErrorID` injected into errors.go becomes exported.

Reply:
It's intentional. So code which uses the library can distinct the error by type cast and/or by checking its id.

Reply:
Oh, yes, of course!

So, we do this on a case by case basis as the need arises? Or is there a convention that I fail to see?

Reply:
It's case by case when there is specific need to make this or the need is foreseeable. For the case of `ReponseServerError`, someone may need to know if error returned from PDP or generated locally by for example inability to transfer data.

Additionally, we have a plan to review and make error handling more formal grounded.

Reply:
Perfect!
Exporting error type on a case by case basis is not ideal, IMHO.
Thanks!